File: articles/from-baghdad-to-latin-europe-algorism-algebra-and-reception/prolog/place_value.pl
place_value(Digits, Value) :-
    place_value_acc(Digits, 0, Value).

place_value_acc([], Acc, Acc).
place_value_acc([Digit|Rest], Acc, Value) :-
    Next is Acc * 10 + Digit,
    place_value_acc(Rest, Next, Value).

% Query example:
% ?- place_value([1,2,3,0], Value).

File: articles/from-baghdad-to-latin-europe-algorism-algebra-and-reception/c/place_value.c
#include <stdio.h>

int main(void) {
    int digits[] = {1, 2, 3, 0};
    int n = sizeof(digits) / sizeof(digits[0]);
    int value = 0;
    for (int i = 0; i < n; i++) value = value * 10 + digits[i];
    printf("place_value=%d\n", value);
    return 0;
}
